### DISCUSSION TITLE
How can I use `nvim-dap` to debug C++?

### DISCUSSION THREAD

**Discussion by vollowx** [ayamir/nvimdots#801] [Q&A] [answered]
As the title says, IDK how to use it.

**Comment by ayamir** [comment, marked as answer]
1. `g++ test.cpp -g`
2. `nvim test.cpp`
3. `<F8>` to toggle breakpoint
4. `<F6>` to run `DapContinue`, fill the prompt content with the executable path and the input args if has.
5. `<C-p>` then input "debug", you can see all of shortcut with debug.
6. Or you can just use the button to control debug.


https://github.com/ayamir/nvimdots/assets/61657399/7fcf4e7f-7c83-40e6-b8f2-c023cc2fc4b1

**Reply by vollowx** [reply]
BTW can I use clang++ with this?

**Reply by ayamir** [reply]
sure, but dont't forget the `-g` argu.

**Reply by vollowx** [reply]
OK, thanks.
